src/java/org/apache/cassandra/cql3/Term.java

```diff
@@ -18,17 +18,18 @@
 package org.apache.cassandra.cql3;
 
 import java.nio.ByteBuffer;
+import java.util.Collections;
 import java.util.List;
 
 import org.apache.cassandra.cql3.functions.Function;
 import org.apache.cassandra.db.marshal.AbstractType;
 import org.apache.cassandra.exceptions.InvalidRequestException;
-import org.apache.cassandra.transport.ProtocolVersion;
+import org.apache.cassandra.schema.SchemaConstants;
 
 /**
  * A CQL3 term, i.e. a column value with or without bind variables.
- *
- * A Term can be either terminal or non terminal. A term object is one that is typed and is obtained
+ * <p>
+ * A Term can be either terminal or non-terminal. A term object is one that is typed and is obtained
```

Comment:
We could put here handy links to the two main subclasses:
```suggestion
 * A Term can be either {@link Terminal} or {@link NonTerminal}. A term object is one that is typed and is obtained
```
Also, if we want to enforce that all subclasses of `Term` are either `Terminal` or `NonTerminal`, we could make this an abstract class with a private constructor. Not sure it is worth the fuzz, though.